bitnami/kubeapps/templates/frontend/deployment.yaml

```diff
@@ -128,6 +128,7 @@ spec:
             - --skip-auth-regex=^\/static\/
             - --skip-auth-regex=^\/$
             - --scope=openid email groups
+            - --cookie-refresh=2m
```

Comment:
A doubt here: how are users expected to modify the default value? Perhaps adding a new `additionalFlags[n]="--cookie-refresh=10s"` ?
If it is not possible or the behavior is not the one user may expect, I'd suggest adding it as a new item in the values.yaml. 

For instance:

```yaml
- --cookie-refresh={{ .Values.authProxy.cookieRefresh }}
```

Reply:
this is the purpose of the change in values.yaml, where i have added --cookie-refresh as a hint in yaml comment for the additionalflags field.

this works exactly as for scope. the deployment.yaml has the default scope, but if you look in values.yaml, the comment for additionalFlags indicates you can override with your own scope.
so users can override with their own --cookie-refresh is they want to.

hope this answers your question.

Reply:
> so users can override with their own --cookie-refresh if they want to.

Thank you for the clarification, but this exactly where I'm not sure about.

I've tried running the chart with these values:

```yaml
authProxy:
  enabled: true
  provider: "test"
  clientID: "test"
  clientSecret: "test"
  cookieSecret: "bm90LWdvb2Qtc2VjcmV0Cg=="
  additionalFlags:
    - --cookie-refresh=blablabla
```

But, as I expected, I'm getting:

```yaml
          args:
            ...
            - --skip-auth-regex=^\/$
            - --scope=openid email groups
            - --cookie-refresh=2m
            - --cookie-refresh=blablabla
```

So, we are generating a duplicated flag here, aren't we?

Reply:
that is correct.
but that is true for the "scope" parameter as well. so this change is following an existing pattern.

Reply:
You are right, have a duplicated `--scope` flag, but it is the way to add multiple scopes, so we are leveraging this feature (as Michael said). 
Conversely, a duplicated `cookie-refresh` flag could introduce unexpected behaviors.
Have you tried adding more than one flag and see what happens?

I strongly suggest going the other way and having just one flag whose value is defined in the values.yaml like ` .Values.authProxy.cookieRefresh`


Now, a question for @absoludity : do you have a reference saying that `scope` can be entered multiple times?
All I'm able to find is:

```
Every command line argument can be specified in a config file by replacing hyphens (-) with underscores (_). If the argument can be specified multiple times, the config option should be plural (trailing s)
```

And, given that `scope` is not `scopeS`, it's not clear to me that we can enter it twice :S

https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview#config-file

Reply:
the "scope" field is not additive, it only takes the last parameter definition.

I ran a small test where i defined the scope int he values.yaml to be only "openid" and thus omitting email and groups.
the authentication failed with the following error from oaut2 proxy:
     Error creating session during OAuth2 callback: id_token did not contain an email and profileURL is not defined

So the behavior is exactly the same for cookie refresh.
instead of changing the pattern for cookie refresh, i think it is better to follow the existing pattern in this issue and create a new issue to change how we handle singleton parameters.

Reply:
Sorry, I wasn't following this PR but had only chatted briefly in the standup about it, and should have checked for context. Let me respond to the different points...

> Now, a question for @absoludity : do you have a reference saying that `scope` can be entered multiple times?

Sorry - my mistake. I was wanting to either test it (as Dimitri has) or check the code to see if it was defined as a [`pflag.StringSlice`](https://pkg.go.dev/github.com/spf13/pflag?utm_source=godoc#StringSlice) which is what I'd thought, but it's not - it's a plain `pflag.String` [as shown in the source](https://github.com/oauth2-proxy/oauth2-proxy/blob/master/pkg/apis/options/legacy_options.go#L555).

> instead of changing the pattern for cookie refresh, i think it is better to follow the existing pattern in this issue and create a new issue to change how we handle singleton parameters.

I think the point is that the existing pattern for `--scope` is (a) not the only existing pattern, and (b) not the best pattern to follow. You'll see in the same file that

```
 - --email-domain={{ .Values.authProxy.emailDomain }}
```

which is exactly the correct pattern in that it allows a useful default that can be overridden in the values and does not result in multiple occurrences of the arg.

So @dlaloue-vmware , do you agree that the pattern used for `--email-domain` makes more sense? If so, let's do that. If not, why not?

Reply:
FWIW, I'd be happy to do the same with `--scope`, either in this PR or separately. I don't remember why it wasn't added as an option in the same way at the time.